internal/store/selector/selector.go

```diff
@@ -235,11 +235,22 @@ func (e *SelectorError) Unwrap() error {
 	return e.OriginalError
 }
 
+// AsSelectorError checks if an error is of type SelectorError and assigns it to target.
+func AsSelectorError(err error, target any) bool {
+	if target == nil {
+		return false
+	}
+	// Ensure target is a pointer to the correct type
+	switch t := target.(type) {
```

Comment:
Hi! I am adding the comment in this line, but it may not the the right place.
The issue is that the API only identifies an invalid field if the first part of the path is incorrect.
eg. "status.summarywhatever.status" is accepted as valid, whereas "statuswhatever.summary.status" is not.

IMO both fields should be identified as invalid fields.

![field-validation](https://github.com/user-attachments/assets/d9546c86-6f89-4630-a6e4-8cd7a77eab7b)

Reply:
@celdrake there's an open issue for it https://issues.redhat.com/browse/EDM-605